pkg/reconciler/subscription/controller.go

```diff
@@ -45,15 +48,18 @@ func NewController(
 
 	subscriptionInformer := subscription.Get(ctx)
 	channelInformer := channel.Get(ctx)
+	serviceaccountInformer := serviceaccountinformer.Get(ctx)
 
 	featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"))
```

Comment:
we should also watch for changes on the feature configmap. So e.g. when the user switches from oidc.authentication: disabled -> enabled, we should reenque to trigger the OIDC service account creation/update.
You can check on https://github.com/knative/eventing/pull/7327#discussion_r1351555450 for an example.

Reply:
@creydr 
Do we still need add configmapinformer , I noticed that you have submitted this pr https://github.com/knative/eventing/pull/7357

Reply:
when you do it via the callback of the feature store, it watches only the feature config map. That is more efficient than using a configmapinformer (and watching all configmaps). That is what we meant in https://github.com/knative/eventing/pull/7327#discussion_r1351555450.

Does this help?